onboard/nav/stateMachine.cpp

```diff
@@ -489,7 +489,7 @@ string StateMachine::stringifyNavState() const
 // Returns true if an obstacle is detected, false otherwise.
 bool StateMachine::isObstacleDetected() const
```

Comment:
Why is this function duplicated with the one in `searchStateMachine`?

Reply:
because the isObstacleDetected() function is a private member function of stateMachine, which is not inherited by gate state machine or searchStateMachine. If you think it's best, I can make it just one public function. it didn't occur to me to change this because the powers that be had implemented it as a private function